sch_fq.c

```diff
@@ -556,13 +785,39 @@ static struct sk_buff *fq_dequeue(struct Qdisc *sch)
 			return NULL;
 		}
 	}
+    }
+
 	f = head->first;
 
+	int rValue = valuePresentinArray(f->socket_hash, pFlowid);
+	
+	printk("rValue is   : %d \n ", rValue);
+	
+	//printk("barrier value  : %d \n ", barrier[dcounter]);
+
+/* Breach and membership of the flow is checked once it is satisfied all the flows are added to co-flow set at once
+	if( (rValue != -1) && (barrier[dcounter] == 3) )
+	{
+	printk("Breach Occured \n");
+	barrier[dcounter] = 0;
+	head->first = f->next;
+	printk("adding all co-flows together \n");
+	Promotecoflows(&q->old_flows,&q->new_flows,&q->co_flows,f, coflow,pFlowid);
+	}
+
+	if(!barrier[dcounter])
+	{
+	   dcounter++;
+	}
```

Comment:
This logic seems a bit muddled because `barrier[.]` is initialized to 0, no? So youre detecting two different conditions with the same value for `barrier[.]`? Why not just set another bit in `barrier[.]` to detect this, in lieu of line 802 above (replacing `barrier[dcounter] = 0`)?